spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -112,6 +112,240 @@ Subsequently, an implementation of Jakarta Data, specifically tailored to the ch
 
 Jakarta Data empowers developers to shape their data access strategies by defining entity classes and repositories, with implementations seamlessly adapting to the chosen data store. This flexibility and Jakarta Data's persistence-agnostic approach promote robust data management within Java applications.
 
+=== Repository interfaces
+
+A Jakarta Data repository is a Java interface annotated `@Repository`.
+A repository interface may declare:
+
+- _abstract_ (non-`default`) methods, and
+- _concrete_ (`default`) methods.
+
+A concrete method may call other methods of the repository, including abstract methods.
+
+Every abstract method of the interface is usually either:
+
+- an entity instance _lifecycle method_,
+- an _annotated query method_,
+- an _automatic query method_, or
+- a _resource accessor method_.
+
+A repository may declare lifecycle methods for a single entity type, or for multiple related entity types.
+Similarly, a repository might have query methods which return different entity types.
+
+A repository interface may inherit methods from a superinterface.
```

Comment:
```suggestion
A non-built-in repository interface may inherit methods from a superinterface.
```

Can we procedure this way? 
WDYT?

Reply:
My vote would be against adding that language because it's true that built-in repository interfaces can inherit methods from a superinterface as well.  This currently exists in the spec where PageableRepository inherits from BasicRepository and inherits its methods as well.  And where CrudRepository inherits from BasicRepository and inherits its methods as well.